commit 73b20257d3090227a1f4ed983a1bbdd7d3367ccf
Author: Volker Krause <vkrause@kde.org>
Date:   Sat Jul 8 10:46:18 2023 +0200

    Fix Android APK generation with Qt 5.15.10
    
    There's two problems here:
    - Gradle OOM's without the changes in gradle.properties
    - Once past that it produces 3x-5x larger APKs due to binary stripping not
      working without specifying the NDK version
    
    This will need to be backported to the release builds as well (in Craft).

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d65aecc1..6c892c84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ include(ECMGenerateQmlTypes)
 
 set(REQUIRED_QT_VERSION "5.15.0")
 if (ANDROID)
-    set(REQUIRED_QT_VERSION "5.15.8")
+    set(REQUIRED_QT_VERSION "5.15.10")
 endif()
 find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Network Qml Quick Test Sql Multimedia Svg Gui Widgets QuickTest Concurrent QuickControls2)
 
diff --git a/data/build.gradle b/data/build.gradle
index 5952c2d3..7f687600 100644
--- a/data/build.gradle
+++ b/data/build.gradle
@@ -44,6 +44,7 @@ android {
     compileSdkVersion androidCompileSdkVersion.toInteger()
 
     buildToolsVersion androidBuildToolsVersion
+    ndkVersion androidNdkVersion
 
     sourceSets {
         main {
diff --git a/data/gradle.properties b/data/gradle.properties
index 066ca7ab..6e98716f 100644
--- a/data/gradle.properties
+++ b/data/gradle.properties
@@ -1,3 +1,7 @@
 # SPDX-FileCopyrightText: 2020 Gabriel Souza Franco <gabrielfrancosouza@gmail.com>
 # SPDX-License-Identifier: BSD-3-Clause
 android.useAndroidX=true
+
+# Gradle properties set by Qt
+org.gradle.jvmargs=-Xmx2500m -XX:MaxMetaspaceSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+org.gradle.parallel=true
